src/overlays/actors/ovl_Bg_Jya_Ironobj/z_bg_jya_ironobj.c

```diff
@@ -43,21 +59,217 @@ static ColliderCylinderInit D_808994AC = {
+    if ((this->dyna.actor.params & 1) == 1) {
+        this->colCylinder.dim.radius = 40;
+        this->colCylinder.dim.height = 100;
+    }
+    Collider_UpdateCylinder(&this->dyna.actor, colCylinder);
+}
+
+/*
+* Spawns particles for the destroyed pillar
+*/
+void BgJyaIronobj_SpawnPillarParticles(BgJyaIronobj* this, GlobalContext* globalCtx, EnIk* enIk) {
+    s32 i;
+    s32 j;
+    s16 unkArg5;
+    f32 temp_f22;
+    Vec3f pos;
+    Vec3f vel;
+    f32 coss;
+    s16 rotY;
+    f32 sins;
+    s32 pad[2];
+
+    if (enIk->unk_2FF <= 0 || enIk->unk_2FF >= 4) {
+        osSyncPrintf("Error 攻撃方法が分からない(%s %d)\n", "../z_bg_jya_ironobj.c", 233, enIk->unk_2FF);
+        return;
+    }
+    osSyncPrintf("¢ attack_type(%d)\n", enIk->unk_2FF);
+    rotY = Actor_WorldYawTowardActor(&this->dyna.actor, &enIk->actor) + D_808994D8[enIk->unk_2FF - 1];
+
+    for (i = 0; i < 8; i++) {
+        Actor* actor =
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_BG_JYA_HAHENIRON, this->dyna.actor.world.pos.x,
+                        Rand_ZeroOne() * 80.0f + this->dyna.actor.world.pos.y + 20.0f, this->dyna.actor.world.pos.z,
+                        0, ((s16)(s32)(Rand_ZeroOne() * 0x4000) + rotY) - 0x2000, 0, 0);
+        if (actor != NULL) {
+            actor->speedXZ = Rand_ZeroOne() * 8.0f + 9.0f;
+            actor->velocity.y = Rand_ZeroOne() * 10.0f + 6.0f;
+        }
+    }
+    Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_BG_JYA_HAHENIRON, this->dyna.actor.world.pos.x,
+                this->dyna.actor.world.pos.y + 150.0f, this->dyna.actor.world.pos.z, 0, 0, 0, 1);
+    Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_BG_JYA_HAHENIRON, this->dyna.actor.world.pos.x,
+                this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z, 0, 0, 0, 2);
+    sins = Math_SinS(rotY);
+    coss = Math_CosS(rotY);
+    for (j = 0; j < 32; j++) {
+        f32 rand = Rand_ZeroOne();
+        if (rand < 0.1f) {
+            unkArg5 = 0x60;
+        } else if (rand < 0.8f) {
+            unkArg5 = 0x40;
+        } else {
+            unkArg5 = 0x20;
+        }
+        pos.x = this->dyna.actor.world.pos.x;
+        pos.y = this->dyna.actor.world.pos.y + ((4.375f * j) + 10.0f);
+        pos.z = this->dyna.actor.world.pos.z;
+        temp_f22 = fabsf(j - 15.5f) * (1.0f / 31) + 0.5f;
+        vel.x = 2.0f * ((Rand_ZeroOne() * 6.0f) - 3.0f) + (Rand_ZeroOne() * sins * 8.0f * temp_f22);
+        vel.y = (Rand_ZeroOne() * 8.0f) - 3.0f;
+        vel.z = 2.0f * ((Rand_ZeroOne() * 6.0f) - 3.0f) + (Rand_ZeroOne() * coss * 8.0f * temp_f22);
+        EffectSsKakera_Spawn(globalCtx, &pos, &vel, &pos, -0x15E, unkArg5, D_80899500[j & 7], 4, 0,
+                             D_808994E0[j & 7], 0, 5, D_808994F0[j & 7], -1, 0x16C, gObjectJyaIronDL_000880);
+        if (Rand_ZeroOne() < 0.26f) {
+            func_80033480(globalCtx, &pos, 200.0f, 1, D_808994E0[j & 7] * 4 + 0x3C, D_808994E0[j & 7] * 4 + 0x50, 1);
```

Comment:
```suggestion
            func_80033480(globalCtx, &pos, 200.0f, 1, D_808994E0[j & 7] * 4 + 60, D_808994E0[j & 7] * 4 + 80, 1);
```
These on the other hand definitely seem decimal.